### PR TITLE
fix(vs-go): Make floating service as option

### PR DESCRIPTION
**Pull request by goock**
The go example for VS makes floating service an option.
Additionally, it upgrades the os image and fixes a few prints.

## TODO:
- [x] - update GitBook

### CODE REVIEW

File: virtual-server/examples/go/main.go [hunk 1]
```diff
@@ -190,44 +197,57 @@ runcmd:
 	// Add the example PVC as a file system to the Virtual Server
 	virtualServer.AddPVCFileSystem("example-storage", pvc.Name, false)
 
-	service := buildFloatingIPService("example-floating-ip-service", namespace)
-	if err := c.Create(context.Background(), service); err != nil {
-		log.Fatalf("Could not create example floatingIP service\nReason: %s", err.Error())
-	}
+	service := &corev1.Service{}
+	if floatingServiceExist {
+		log.Printf("Creating floating service %s", floatingServiceName)
```
Comment (YitzyD) [inline virtual-server/examples/go/main.go:202]: "Creating floatingIP service"

File: virtual-server/examples/go/main.go [hunk 2]
```diff
@@ -190,44 +197,57 @@ runcmd:
...
+			log.Fatalf("Could not create example floatingIP service, reason: %s", err.Error())
+		}
 
-	// Add the example floatingIP service to the VirtualServer
-	virtualServer.AddFloatingIP(service.Name)
+		// Add the example floatingIP service to the VirtualServer
+		virtualServer.AddFloatingIP(service.Name)
+	} else {
+		log.Print("Floating service disabled")
+	}
```
Comment (YitzyD) [inline virtual-server/examples/go/main.go:212]: Probably don't need this else block. Not really useful info if they didn't specify

**Review by YitzyD — CHANGES_REQUESTED** [review]

File: virtual-server/examples/go/main.go [hunk 3]
```diff
@@ -139,12 +146,12 @@ func main() {
 	// sourced in the vd-images namespace
 	err = virtualServer.ConfigureStorageRootWithPVCSource(vsv1alpha.VirtualServerStorageRootPVCSource{
 		Size:             "40Gi",
-		PVCName:          "ubuntu1804-nvidia-465-19-01-1-docker-master-20210629-ord1",
+		PVCName:          "ubuntu2004-nvidia-470-103-01-1-docker-master-20220215-ord1",
```
[the remark this answers is not in the document]
Comment (goock) [reply virtual-server/examples/go/main.go]: lol, IT is the quickest changing domain of our life

File: virtual-server/examples/go/main.go (hunk 2 above)
Comment (goock) [reply virtual-server/examples/go/main.go]: removed

**Comment by goock** [comment]
@microcosem Probably you want to take a look at the `README.md` file and apply the changes to GitBook.

**Comment by salanki** [comment]
I feel like floating IP should play less central role in this example. It is a feature used by no-one, and we don't particularly like it. I'm fine keeping it around somewhere, but there are a gazillion features that should be more prominent than floating IP in the Go example.

**Comment by microcosem** [comment]
@salanki I've adjusted the docs to reflect that the example does illustrate floating IP creation, but made it less prominent a feature in the example doc.

**Review by microcosem — APPROVED** [review]

**Pull request merged**